### ISSUE TITLE
How to pass argc and argv to C FFI ?

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang-nursery/rust-cookbook#504]
Many libraries require forwarding them `argc` and `argv` from Rust, ideally, without allocating any memory. 

I thought this would make for a good "cookbook" example.

**Comment by lwshang** [comment]
Is there any blog/post on this topic?

**Comment by KappaDistributive** [comment]
Right now, there isn't really a chapter on [Rust's FFI](https://doc.rust-lang.org/nomicon/ffi.html) in the cookbook and I'm wondering whether adding this (and related) examples would overreach the goals of the current treatment of the FFI in the cookbook. I.e. I think, if we decide to pursue this goal, we should create a dedicated chapter on the FFI first.

That being said, in a previous discussion ( #328 ) people seem to have come to the conclusion that an FFI chapter in the cookbook might be an unwise idea as presenting it in the form of a small collection of snippets might ultimately prove to be of disservice to the reader. (The delightful term 'footgun' has come up many times in that discussion.)

I'd be happy to reevaluate the discussion whether the cookbook should have an FFI chapter but I also think that, if we decide to include it, we should take the warnings of #328 to heart and have a discussion how to set it up in a reasonable, responsible way rather than including random snippets and hoping for the best.

**Comment by gnzlbg** [comment]
Isn't there a Rust FFI book somewhere? That might be something worth pursuing instead.

**Comment by KappaDistributive** [comment]
@gnzlbg There is a chapter about [FFI in the Rust book](https://doc.rust-lang.org/book/ffi.html) if that's what you meant.

**Comment by gnzlbg** [comment]
No, I meant a book whose only purpose is to cover how to use Rust FFI features to interface with the rest of the world (e.g. this issue would just be a tiny part of such a book).

**Comment by AndyGauge** [comment]
There's the [nomicon](https://doc.rust-lang.org/nomicon/ffi.html) and [The unofficial Rust FFI Guide](https://michael-f-bryan.github.io/rust-ffi-guide/overview.html)
